support/cas-server-support-oidc-core-api/src/main/java/org/apereo/cas/oidc/services/OidcServicesManagerRegisteredServiceLocator.java

```diff
@@ -1,23 +1,13 @@
 package org.apereo.cas.oidc.services;
 
-import org.apereo.cas.services.DefaultServicesManagerRegisteredServiceLocator;
-import org.apereo.cas.services.OidcRegisteredService;
-import org.apereo.cas.support.oauth.OAuth20Constants;
-
-import org.springframework.core.Ordered;
+import org.apereo.cas.support.oauth.services.OAuth20ServicesManagerRegisteredServiceLocator;
 
 /**
  * This is {@link OidcServicesManagerRegisteredServiceLocator}.
  *
  * @author Misagh Moayyed
  * @since 6.3.0
  */
-public class OidcServicesManagerRegisteredServiceLocator extends DefaultServicesManagerRegisteredServiceLocator {
-    public OidcServicesManagerRegisteredServiceLocator() {
-        setOrder(Ordered.HIGHEST_PRECEDENCE);
-        setRegisteredServiceFilter((registeredService, service) -> service.getAttributes().containsKey(OAuth20Constants.CLIENT_ID)
-            && registeredService.getClass().equals(OidcRegisteredService.class)
-            && registeredService.matches(service.getId()));
-    }
+public class OidcServicesManagerRegisteredServiceLocator extends OAuth20ServicesManagerRegisteredServiceLocator {
```

Comment:
Another option is that this locator could be removed completly.

Reply:
This looks good to me in general. I would however suggest that you repeat the same condition for OAUTH but the small adjustment that the locator would operate on OIDC services only for a few reasons: easier to port forward, easier for extensions and customizations of oauth/oidc services and their own locators, and while quite rare, it's possible for an oauth client and oidc client to share the same client id, given they are from distinct protocols per se.